ide/tasks/git.py

```diff
@@ -83,17 +81,23 @@ def github_push(user, commit_message, repo_name, project):
         root = find_project_root(paths)
     except:
         root = ''
+    expected_paths = set()
+
+    def update_expected_paths(new_path):
+        # This adds the path *and* all parent directories to the list of expected paths.
```

Comment:
The directories themselves are in next_tree, so if you don't add them to expected_paths then, when you iterate over next_tree looking for paths which aren't in expected_paths (which represent files to be deleted), it will look as if the directories have been deleted, so has_changed will get set to True.

An alternative to this would be to check (in the "#manage deleted files" block) if the path in next_tree looks like a directory, and if it is, ignore it. However, then you'd need a way to check if a directory actually *was* deleted (e.g. if someone deleted all their workers).